File: tests/fixtures/user_controller.py
"""Example controller module used as a Stage 2 parsing fixture."""
import json
from typing import Optional

from services.user_service import UserService
from db.models.orm import UserRecord as UserRow


class UserController:
    def __init__(self, service: UserService):
        self.service = service

    def get_user(self, user_id: str) -> Optional[dict]:
        user = self.service.find_by_id(user_id)
        if user is None:
            return None
        return json.loads(user.to_json())

    def create_user(self, payload: dict) -> dict:
        return self.service.create(payload)


def health_check() -> dict:
    return {"status": "ok"}

File: tests/fixtures/user_controller.ts
import { UserService } from "../services/userService";
import * as express from "express";

export function healthCheck(): object {
  return { status: "ok" };
}

export class UserController {
  private service: UserService;

  constructor(service: UserService) {
    this.service = service;
  }

  getUser(id: string): object | null {
    return this.service.findById(id);
  }

  createUser(payload: object): object {
    return this.service.create(payload);
  }
}
